Systems/c172p-engine.xml

```diff
@@ -62,7 +62,101 @@
             </function>
             <output>/fdm/jsbsim/propulsion/engine[1]/air-intake-impedance-factor</output>
         </fcs_function>
-        
+
+    </channel>
+
+    <channel name="damage">
+        <fcs_function name="RPM over Redline">
+            <function>
+                <sum>
+                    <property>/engines/active-engine/rpm</property>
+                    <value>-2700</value>
+                </sum>
+            </function>
+            <clipto>
+                <min>0</min>
+                <max>10000</max>
+            </clipto>
+            <output>/engines/active-engine/rpm-over-redline</output>
+        </fcs_function>
+
+        <fcs_function name="damage level calculation">
+              <function>
+                <ifthen>
+                    <eq>
+                        <p>/engines/active-engine/damage_allowed</p>
+                        <v>1</v>
+                    </eq>
+                    <sum>
+                        <property>/engines/active-engine/damage-level</property>
+                        <product>
+                           <table>
+                                <independentVar lookup="row">/engines/active-engine/rpm-over-redline</independentVar>
+                                <tableData>
+                                    0    0
+                                  100    1
+                                  200    4
+                                  300    9
+                                  400   16
+                                  500   25
+                                  600   36
+                                  700   49
+                                  800   64
+                                  900   81
+                                 1000  100
+                                 1100  121
+                                 1200  144
+                                 1300  169
+                                 1400  196
+                                 1500  225
+                                 9999  300
+                                </tableData>
+                            </table>
+                            <property>simulation/channel-dt</property>
+                        </product>
+                    </sum>
+                    <v>0</v> <!-- this is the value if damage is disabled -->
+                </ifthen>
+            </function>
+            <output>/engines/active-engine/damage-level</output>
+        </fcs_function>
+
+        <fcs_function name="volumetric factor calculation">
+            <function>
+				<difference>
+					<v>1.0</v>
+					<product>
+						<property>/engines/active-engine/damage-level</property>
+						<v>0.00166</v>
+					</product>
+				</difference>
+            </function>
+            <clipto>
+                <min>0.0</min>
+                <max>1.0</max>
+            </clipto>
+            <output>/engines/active-engine/damage-volumetric-efficiency-factor</output>
```

Comment:
Is this property only used in this file? Then you don't need this `<output>`. You can just use `fcs/volumetric-factor-calculation` in the fcs_function below.

These `<output>` properties are mostly only needed when it is used by Nasal listeners for example.